watch-parties/src/test/java/com/google/sps/servlets/VideoSyncServletTest.java

```diff
@@ -29,35 +29,54 @@
 /** Class to test VideoSyncServlet. */
 @RunWith(JUnit4.class)
 public final class VideoSyncServletTest {
-  private HttpServletRequest mockRequest;       
-  private HttpServletResponse mockResponse; 
-
-  @Before
-  public void setUp() {
-    mockRequest = mock(HttpServletRequest.class);       
-    mockResponse = mock(HttpServletResponse.class);   
-  }
-
-  @Test
-  public void testDoGet() throws IOException {
-    StringWriter stringWriter = new StringWriter();
-    PrintWriter writer = new PrintWriter(stringWriter);
-    when(mockResponse.getWriter()).thenReturn(writer);
-
-    VideoSyncServlet servlet = new VideoSyncServlet();
-    servlet.videoStatus = 200;
-    servlet.doGet(mockRequest, mockResponse);
-
-    writer.flush();
-    Assert.assertTrue(stringWriter.toString().contains("200"));
-  }
-
-  @Test
-  public void testDoPost() throws IOException {
-    when(mockRequest.getParameter("status")).thenReturn("100");
-
-    VideoSyncServlet servlet = new VideoSyncServlet();
-    servlet.doPost(mockRequest, mockResponse);
-    Assert.assertEquals(100, servlet.videoStatus); 
-  }
+    private HttpServletRequest HostMockRequest;       
+    private HttpServletResponse HostMockResponse; 
+    private HttpServletRequest UserMockRequest;       
+    private HttpServletResponse UserMockResponse; 
+
+    @Before
+    public void setUp() {
+        // Create mock and request with host privileges
+        HostMockRequest = mock(HttpServletRequest.class);      
+        when(HostMockRequest.getParameter("host")).thenReturn("true"); 
+        HostMockResponse = mock(HttpServletResponse.class);   
+
+        // Create mock and request with regular privileges
+        UserMockRequest = mock(HttpServletRequest.class);       
+        UserMockResponse = mock(HttpServletResponse.class);   
+    }
+
+    @Test
+    public void testDoGet() throws IOException {
+        StringWriter stringWriter = new StringWriter();
+        PrintWriter writer = new PrintWriter(stringWriter);
+        when(UserMockResponse.getWriter()).thenReturn(writer);
+
+        VideoSyncServlet servlet = new VideoSyncServlet();
+        servlet.videoStatus = 200;
+        servlet.doGet(UserMockRequest, UserMockResponse);
+
+        writer.flush();
+        Assert.assertTrue(stringWriter.toString().contains("200"));
+    }
+
+    @Test
+    public void testDoPostHostTrue() throws IOException {
```

Comment:
Let's rename this function (and the one below) so that it clearly specifies the purpose of the test.

may testDoPostOnHostUser and testDoPostOnGeneralUser or something like that.